pkg/operator/ceph/object/spec.go

```diff
@@ -540,8 +576,34 @@ func (c *clusterConfig) generateVolumeSourceWithTLSSecret() (*v1.SecretVolumeSou
 	return secretVolSrc, nil
 }
 
-func (c *clusterConfig) rgwTLSSecretType() (v1.SecretType, error) {
-	rgwTlsSecret, err := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(context.TODO(), c.store.Spec.Gateway.SSLCertificateRef, metav1.GetOptions{})
+func (c *clusterConfig) generateVolumeSourceWithCaBundleSecret() (*v1.SecretVolumeSource, error) {
+	// Keep the ca-bundle as secure as possible in the container. Give only user read perms.
+	// Same as above for generateVolumeSourceWithTLSSecret function.
+	userReadOnly := int32(0400)
+	var caBundleVolSrc *v1.SecretVolumeSource
+	// TODO: add support for OpenShift based on ConfigMapVolumeSource and key `service.beta.openshift.io/inject-cabundle`
+	if c.store.Spec.Gateway.CaBundleRef != "" {
```

Comment:
The caller already checked for this, so it's a safe assumption that it's set and we can simplify the error checking here.
```suggestion
```